include/message_filters/subscriber.h

```diff
@@ -68,6 +70,47 @@ class SubscriberBase
    * \param qos (optional) The rmw qos profile to use to subscribe
    */
   virtual void subscribe(rclcpp::Node * node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default) = 0;
+
+  /**
+   * \brief Subscribe to a topic.
+   *
+   * If this Subscriber is already subscribed to a topic, this function will first unsubscribe.
+   * This override allows SubscriptionOptions to be passed into the class without changing API.
+   *
+   * \param node The rclcpp::Node::SharedPtr to use to subscribe.
+   * \param topic The topic to subscribe to.
+   * \param qos (optional) The rmw qos profile to use to subscribe.
+   * \param options The subscription options to use to subscribe.
+   */
+  virtual void subscribe(
+    rclcpp::Node::SharedPtr node,
+    const std::string& topic,
+    const rmw_qos_profile_t qos,
+    rclcpp::SubscriptionOptions options)
+  {
+    this->subscribe(node.get(), topic, qos, options);
+  };
+
+  /**
+   * \brief Subscribe to a topic.
+   *
+   * If this Subscriber is already subscribed to a topic, this function will first unsubscribe.
+   *
+   * \param node The rclcpp::Node to use to subscribe.
+   * \param topic The topic to subscribe to.
+   * \param qos (optional) The rmw qos profile to use to subscribe
+   */
+  virtual void subscribe(
+    rclcpp::Node * node,
+    const std::string& topic,
+    const rmw_qos_profile_t qos,
+    rclcpp::SubscriptionOptions options)
+  {
+    (void) options;
+    RCLCPP_WARN(node->get_logger(), "SubscriberBase::subscribe with four arguments has not been overridden");
```

Comment:
On second thought, I think this should be `RCLCPP_ERROR`, since it's clearly a bug if someone calls this and the `options` are ignored. Sorry for changing my mind...

Reply:
No worries